airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -86,6 +89,19 @@ def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapp
                 self.logger.warning(f"Unexpected error in `parse_ersponse`: {e}, the actual response data: {response.text}")
                 yield {}
 
+    def read_records(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        **kwargs,
+    ) -> Iterable[Mapping[str, Any]]:
+        """Override to fetch deleted records for supported streams"""
+        # main records stream
+        yield from super().read_records(stream_state=stream_state, stream_slice=stream_slice, **kwargs)
+        # fetch deleted events after the Stream data is pulled
+        if self.supports_deleted_events:
+            yield from self.deleted_events_instance.read_records(stream_state=stream_state, **kwargs)
```

Comment:
this code structure does not guarantee `deleted_events_instance` won't be undefined, if this class expects that prop to exist then it should guarantee it somehow via `@abstractmethod` or something else

Reply:
got rid of this dependency now, it's guaranteed by the class usage itself for target streams to support the events fetching, The class called `IncrementalShopifyStreamWithDeletedEvents` now has this property inside., thus automatically used by the sub-classes.